atomic_reactor/plugin.py

```diff
@@ -219,17 +230,36 @@ def run(self, keep_going=False):
             logger.debug("running plugin '%s'", plugin_name)
             start_time = datetime.datetime.now()
 
+            plugin_response = None
+            skip_response = False
             try:
                 plugin_instance = self.create_instance_from_plugin(plugin_class, plugin_conf)
                 self.save_plugin_timestamp(plugin_class.key, start_time)
                 plugin_response = plugin_instance.run()
+                plugin_successful = True
+                if buildstep_phase:
+                    assert isinstance(plugin_response, BuildResult)
+                    if plugin_response.is_failed():
+                        msg = "Build step plugin %s failed" % plugin_class.key
+                        logger.error(msg)
+                        self.on_plugin_failed(plugin_class.key, "Buildstep plugin failed")
+                        plugin_successful = False
+                        self.plugins_results[plugin_class.key] = plugin_response
+                        break
+
             except AutoRebuildCanceledException as ex:
                 # if auto rebuild is canceled, then just reraise
                 # NOTE: We need to catch and reraise explicitly, so that the below except clause
                 #   doesn't catch this and make PluginFailedException out of it in the end
                 #   (calling methods would then need to parse exception message to see if
                 #   AutoRebuildCanceledException was raised here)
                 raise
+            except InappropriateBuildStepError:
+                logger.debug('Build step %s is not appropriate', plugin_class.key)
+                # don't put None, in results for InappropriateBuildStepError
+                skip_response = True
+                if not buildstep_phase:
+                    raise
```

Comment:
This `raise` doesn't get test coverage, but it looks ok on visual inspection. ;)